client/src/modules/cohort/components/CohortHeader.jsx

```diff
@@ -48,43 +48,49 @@ class CohortHeader extends PureComponent {
       isNameInputVisible,
       nameInputValue
     } = this.state;
+    const {
+      details: { name }
+    } = this.props;
 
     if (isDescriptionTextareaVisible && isClickedOutside) {
       this.handleDescriptionUpdate();
       return;
     }
 
-    if (
-      isNameInputVisible &&
-      nameInputValue.trim().length >= 1 &&
-      isClickedOutside
-    ) {
-      this.handleNameUpdate();
+    if (isNameInputVisible && isClickedOutside) {
+      if (_trim(nameInputValue).length >= 1) {
+        return this.handleNameUpdate();
+      }
+      this.setState({ isNameInputVisible: false, nameInputValue: name });
     }
   };
 
   handleKeyPress = event => {
     const { isDescriptionTextareaVisible } = this.state;
     const { code } = event;
 
-    if (code === KeyCodes.ENTER || code === KeyCodes.ESCAPE) {
+    if (code === KeyCodes.ESCAPE) {
       const action = isDescriptionTextareaVisible
         ? this.handleDescriptionUpdate
         : this.handleNameUpdate;
 
       action();
     }
```

Comment:
should return as if the `code === ESCAPE` then we don't need to check other values